DQMOffline/Trigger/test/testHLTFiltersDQMonitor_cfg.py

```diff
@@ -0,0 +1,105 @@
+import FWCore.ParameterSet.Config as cms
+
+## CLI parser
+import argparse
+import sys
+
+parser = argparse.ArgumentParser(
+    prog = 'cmsRun '+sys.argv[0]+' --',
+    description = 'Configuration file to test of the HLTFiltersDQMonitor plugin.',
+    formatter_class = argparse.ArgumentDefaultsHelpFormatter
+)
+
```

Comment:
Just for completeness: this `argparse` configuration was ~copied from
https://github.com/cms-sw/cmssw/blob/287f4db20c2fce1650201e261bb21edc7a8099ee/FWCore/Integration/test/testSwitchProducerTask_cfg.py#L3-L14
I was wondering if there are cases where `VarParsing` has advantages (or, is recommended) wrt `argparse`.

The only hiccup I encountered with this `argparse` was with
```sh
cmsRun DQMOffline/Trigger/test/testHLTFiltersDQMonitor_cfg.py -- -h
```
which I naively expected to work, but leads to an error [1] (didn't figure out the solution).

[1]
```
----- Begin Fatal Exception 04-Jan-2023 19:08:48 CET-----------------------
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named DQMOffline/Trigger/test/testHLTFiltersDQMonitor_cfg.py
Exception Message:
 unknown python problem occurred.
SystemExit: 0

At:
  /cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc11/external/python3/3.9.14-76a14295dd5255228210eb596893b98c/lib/python3.9/argparse.py(2569): exit
  /cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc11/external/python3/3.9.14-76a14295dd5255228210eb596893b98c/lib/python3.9/argparse.py(1100): __call__
  /cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc11/external/python3/3.9.14-76a14295dd5255228210eb596893b98c/lib/python3.9/argparse.py(1935): take_action
  /cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc11/external/python3/3.9.14-76a14295dd5255228210eb596893b98c/lib/python3.9/argparse.py(2007): consume_optional
  /cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc11/external/python3/3.9.14-76a14295dd5255228210eb596893b98c/lib/python3.9/argparse.py(2067): _parse_known_args
  /cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc11/external/python3/3.9.14-76a14295dd5255228210eb596893b98c/lib/python3.9/argparse.py(1861): parse_known_args
  DQMOffline/Trigger/test/testHLTFiltersDQMonitor_cfg.py(37): <module>

----- End Fatal Exception -------------------------------------------------
```

Reply:
> Just for completeness: this `argparse` configuration was ~copied from

For future reference the use of `argparse` is also described in
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideAboutPythonConfigFile#argparse

> The only hiccup I encountered with this `argparse` was with
> 
> ```shell
> cmsRun DQMOffline/Trigger/test/testHLTFiltersDQMonitor_cfg.py -- -h
> ```
> 
> which I naively expected to work, but leads to an error [1]

Hah, interesting. It seems that `sys.exit(0)` gets communicated as a thrown exception by PyBind11 in the C++ side, which framework then reports as shown (the printout from `-h` is there, although in a slightly confusing way). It is not immediately clear to me if we should do something about it, so a specific issue _might_ be worth it.

As for `VarParsing` vs `argparse`, `core` does not have a specific recommendation. First has longer history within CMSSW (and not really developed further), and the latter comes from python's standard library.